math/sum_of_digits.cpp

```diff
@@ -1,72 +1,74 @@
 /**
- * Copyright 2020 @author iamnambiar
- *
  * @file
- * \brief A C++ Program to find the Sum of Digits of input integer.
+ * @brief
+ * A C++ Program to find the [Digit
+ * Sum](https://en.wikipedia.org/wiki/Digit_sum) of any given number.
+ * @details A Digit Sum is found by adding the digits of a number.
+ * @author [iamnambiar](https://github.com/iamnambiar)
+ * @author [Dylan Cusson](https://github.com/dylancusson)
  */
-#include <cassert>
-#include <iostream>
+#include <cassert>   //Include for asserts
+#include <iostream>  //Include for IO operations
 
 /**
- * Function to find the sum of the digits of an integer.
+ * @brief Mathematical algorithms
+ * @namespace
+ */
+namespace math {
+/**
+ * @brief Function to find the sum of the digits of an integer.
  * @param num The integer.
  * @return Sum of the digits of the integer.
- *
- * \detail
- * First the algorithm check whether the num is negative or positive,
- * if it is negative, then we neglect the negative sign.
- * Next, the algorithm extract the last digit of num by dividing by 10
- * and extracting the remainder and this is added to the sum.
- * The number is then divided by 10 to remove the last digit.
- * This loop continues until num becomes 0.
  */
 int sum_of_digits(int num) {
-    // If num is negative then negative sign is neglected.
+    /*
+     * First the algorithm checks whether the number is negative or positive,
+     * and sets the sign accordingly.
```

Comment:
the answer should always be positive no?

Reply:
https://en.wikipedia.org/wiki/Digit_sum
Was based on the wikipedia article which allows for negative digits, though I suppose with that interpretation then the whole `sign` and `num < 0` business is unnecessary.